radixdlt-engine/src/main/java/com/radixdlt/application/system/scrypt/EpochUpdateConstraintScrypt.java

```diff
@@ -278,12 +305,7 @@ public ReducerState bootUp(ValidatorBFTData data) throws MismatchException {
     }
   }
 
-  public static class StartingNextEpoch implements ReducerState {
-    private final HasEpochData prevEpoch;
-
-    StartingNextEpoch(HasEpochData prevEpoch) {
-      this.prevEpoch = prevEpoch;
-    }
+  public record StartingNextEpoch(HasEpochData prevEpoch) implements ReducerState {
```

Comment:
Didn't know nested records are implicitly static.

Reply:
This should be expected, given that all elements of record should be explicitly declared. Hence, there is no way to provide reference to parent object.